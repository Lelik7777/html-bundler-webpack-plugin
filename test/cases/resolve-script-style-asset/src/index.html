<html>
  <head>
    <title>Test</title>

    <link href="./apple.png">
    <link imagesrcset="./apple.png" rel="preload" as="image">

    <link href="./favicon.ico" rel="icon" />
    <link href="./style.css" type="text/css" />
    <link href="./base.css" rel="alternate stylesheet" />
    <link href="./open-sans-regular.woff2" rel="preload" as="font" type="font/woff2" />
    <script src="./main.js"></script>
  </head>
  <body>
    <h1>Hello World!</h1>

    <img src="./apple.png">

    <!-- tag w/o any attributes -->
    <img>
    <img />

    <!-- tag w/o source attribute -->
    <img alt="apple">

    <!-- tag with source attribute only -->
    <img src="./apple.png">

    <!-- w/o space between attributes -->
    <img src="./fig1.png"title="fig">

    <!-- ignore resolving of src attribute when leading char is not a space symbol
         there should be a space between attribute and previous attribute -->
    <img title="fig"src="./fig2.png">

    <!-- test tags sort -->
    <picture>
      <source srcset="./fig1.png, ./fig2.png 100w, ./fig3.png 1.5x">
    </picture>

    <!-- test attributes sort -->
    <img srcset="./lime1.png, ./lime2.png 100w, ./lime3.png 1.5x" src="./lime.png">

    <!-- space around values in srcset attribute -->
    <img srcset=" ./fig1.png  , ./fig2.png  100w  ,  ./fig3.png 1.5x " src=" ./lime.png ">

    <!-- w/o space around values in srcset attribute -->
    <img srcset="./lime1.png,./lime2.png 100w,./lime3.png 1.5x" src="./lime.png">

    <!-- multiline attributes -->
    <img
      alt="apple"
      src="./apple.png"
    >

    <!-- multiline attributes -->
    <img srcset="./lime1.png,
                 ./lime2.png 100w,
                 ./lime3.png 1.5x"
         src="./lime.png">

    <div class="logo">
      <img src="./apple.png" alt="apple">
    </div>

    <!-- image with query -->
    <img src="./apple.png?foo=bar" alt="apple">

    <picture>
      <source srcset="./plum.png" type="image/png" />
    </picture>

    <input src="./apple.png" type="image">

    <object width="200" height="150" data="./apple.png"></object>

    <p>Ignore resources</p>

    <!-- bypass url -->
    <img src="https://example.com/img/image.png">
    <img src="http://example.com/img/image.png">
    <img src="ftp://example.com/img/image.png">
    <img src="//example.com/img/image.png">
    <img src="/img/image.png">

    <!-- bypass data Url -->
    <img src="data:image/png;base64,iVBORw0KGgoAAAA==">

    <!-- bypass invalide source file -->
    <img src="#">
    <img src="javascript:alert('hello');">
    <img src="mailto:admin@test.com">
  </body>
</html>