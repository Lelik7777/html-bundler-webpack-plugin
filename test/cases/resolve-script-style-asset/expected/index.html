<html>
  <head>
    <title>Test</title>

    <link href="assets/img/apple.697ef306.png">
    <link imagesrcset="assets/img/apple.697ef306.png" rel="preload" as="image">

    <link href="assets/img/favicon.697ef306.ico" rel="icon" />
    <link href="assets/css/style.026fd625.css" type="text/css" />
    <link href="assets/css/base.e58351d5.css" rel="alternate stylesheet" />
    <link href="assets/fonts/open-sans-regular.woff2" rel="preload" as="font" type="font/woff2" />
    <script src="assets/js/main.5a6d9860.js"></script>
  </head>
  <body>
    <h1>Hello World!</h1>

    <img src="assets/img/apple.697ef306.png">

    <!-- tag w/o any attributes -->
    <img>
    <img />

    <!-- tag w/o source attribute -->
    <img alt="apple">

    <!-- tag with source attribute only -->
    <img src="assets/img/apple.697ef306.png">

    <!-- w/o space between attributes -->
    <img src="assets/img/fig1.697ef306.png"title="fig">

    <!-- ignore resolving of src attribute when leading char is not a space symbol
         there should be a space between attribute and previous attribute -->
    <img title="fig"src="./fig2.png">

    <!-- test tags sort -->
    <picture>
      <source srcset="assets/img/fig1.697ef306.png, assets/img/fig2.697ef306.png 100w, assets/img/fig3.697ef306.png 1.5x">
    </picture>

    <!-- test attributes sort -->
    <img srcset="assets/img/lime1.697ef306.png, assets/img/lime2.697ef306.png 100w, assets/img/lime3.697ef306.png 1.5x" src="assets/img/lime.697ef306.png">

    <!-- space around values in srcset attribute -->
    <img srcset=" assets/img/fig1.697ef306.png  , assets/img/fig2.697ef306.png  100w  ,  assets/img/fig3.697ef306.png 1.5x " src="assets/img/lime.697ef306.png">

    <!-- w/o space around values in srcset attribute -->
    <img srcset="assets/img/lime1.697ef306.png,assets/img/lime2.697ef306.png 100w,assets/img/lime3.697ef306.png 1.5x" src="assets/img/lime.697ef306.png">

    <!-- multiline attributes -->
    <img
      alt="apple"
      src="assets/img/apple.697ef306.png"
    >

    <!-- multiline attributes -->
    <img srcset="assets/img/lime1.697ef306.png,
                 assets/img/lime2.697ef306.png 100w,
                 assets/img/lime3.697ef306.png 1.5x"
         src="assets/img/lime.697ef306.png">

    <div class="logo">
      <img src="assets/img/apple.697ef306.png" alt="apple">
    </div>

    <!-- image with query -->
    <img src="assets/img/apple.697ef306.png?foo=bar" alt="apple">

    <picture>
      <source srcset="assets/img/plum.697ef306.png" type="image/png" />
    </picture>

    <input src="assets/img/apple.697ef306.png" type="image">

    <object width="200" height="150" data="assets/img/apple.697ef306.png"></object>

    <p>Ignore resources</p>

    <!-- bypass url -->
    <img src="https://example.com/img/image.png">
    <img src="http://example.com/img/image.png">
    <img src="ftp://example.com/img/image.png">
    <img src="//example.com/img/image.png">
    <img src="/img/image.png">

    <!-- bypass data Url -->
    <img src="data:image/png;base64,iVBORw0KGgoAAAA==">

    <!-- bypass invalide source file -->
    <img src="#">
    <img src="javascript:alert('hello');">
    <img src="mailto:admin@test.com">
  </body>
</html>